src/Microsoft.DocAsCode.Common/Loggers/AsyncLogListener.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.Common
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    public class AsyncLogListener : ILoggerListener
+    {
+        private BlockingCollection<ILogItem> _logQueue = new BlockingCollection<ILogItem>();
+        private readonly ManualResetEvent _signal = new ManualResetEvent(true);
+        private readonly ManualResetEvent _flushSignal = new ManualResetEvent(true);
+        private readonly IEnumerable<ILoggerListener> _innerListeners;
+        private readonly int TimeoutMilliseconds = 300000; // 5 minutes
+
+        public AsyncLogListener(IEnumerable<ILoggerListener> listeners)
+        {
+            if (listeners == null)
+            {
+                throw new ArgumentNullException(nameof(listeners));
+            }
+
+            _innerListeners = listeners;
+            Task.Factory.StartNew(LoggingTask);
+        }
+
+        public void WriteLine(ILogItem item)
+        {
+            _signal.Reset();
+            _logQueue.Add(item);
+        }
+
+        public void Flush()
+        {
+            _signal.WaitOne(TimeoutMilliseconds);
+            foreach (var listener in _innerListeners)
+            {
+                listener.Flush();
+            }
+        }
+
+        public void Dispose()
+        {
+            _signal.WaitOne(TimeoutMilliseconds);
+            foreach (var listerner in _innerListeners)
+            {
+                listerner.Dispose();
+            }
+            _signal.Dispose();
+            _logQueue.Dispose();
+        }
+
+        private void LoggingTask()
+        {
+            while (true)
+            {
+                if (_logQueue.Count == 0)
+                {
+                    _signal.Set();
+                }
+
+                ILogItem item = _logQueue.Take();
```

Comment:
>Take [](start = 42, length = 4)

it will throw object disposed.
[https://msdn.microsoft.com/en-us/library/dd287085(v=vs.110).aspx](https://msdn.microsoft.com/en-us/library/dd287085(v=vs.110).aspx)